### ISSUE TITLE
Generating nested routes while trailingSlash is true

### ISSUE THREAD

**Issue by cain** [nuxt/nuxt.js#8165]
### Versions

- nuxt: 12.14.6

### Reproduction
Visit the link and inside sandbox terminal run: `npm run generate`
https://codesandbox.io/s/nuxt-generate-trailing-slash-bug-x7bhd?file=/nuxt.config.js

### Steps to reproduce
1. Setup nested routes https://nuxtjs.org/guide/routing#nested-routes
```
/pages/about.vue
/pages/about/company.vue # or /pages/about/company/index.vue
```
2. Enable `trailingSlash: true` inside nuxt.config
2. Run command `nuxt generate` to generate nested pages

### What is Expected?
Success generation of nested routes containing trailing slash.
```
✔ Generated route "/about/company/"
✔ Generated route "/about/info/"  
```

### What is actually happening?
The command is failing with error `This page could not be found` and missing trailing slashes on nested routes.

Note: the build actually completes without issues, just seems to be the build error.
```
Error generating route "/about/company": This page could not be found
Error generating route "/about/info": This page could not be found
```

![Kapture 2020-10-08 at 17 29 37](https://user-images.githubusercontent.com/15246256/95422937-f4607b80-098b-11eb-8c79-b985e0f40a9e.gif)

**Comment by cain** [comment]
Please let me know if im missing anything here, also if you have any clue why the nested routes contain no forward slash

**Comment by mahapo** [comment]
@cain +1 Trailing slash are needed for netflify

**Comment by mornir** [comment]
Yes! This is a bug! (which is actually is regression, before 2.14.0 it was working fine). Reporting this bug was on my todo list for awhile. Thanks for reporting and opening a pull request!
(With fallback set to true, those not found routes are still accessible. Or you can manually add them in routes.)

**Comment by pi0** [comment]
This should be fixed in [v2.14.8](https://github.com/nuxt/nuxt.js/releases/tag/v2.14.8). Please reopen if not.

**Comment by mornir** [comment]
I've just upgraded to v.2.14.8 and my nested routes are correctly generated!😃 Thanks everyone for the fix!👍
